src/bbs2gh/Commands/MigrateRepoCommand.cs

```diff
@@ -2,141 +2,239 @@
 using System.CommandLine;
 using System.CommandLine.Invocation;
 using System.Threading.Tasks;
+using OctoshiftCLI.Extensions;
 
-namespace OctoshiftCLI.BbsToGithub.Commands
+namespace OctoshiftCLI.BbsToGithub.Commands;
+
+public class MigrateRepoCommand : Command
 {
-    public class MigrateRepoCommand : Command
+    private readonly OctoLogger _log;
+    private readonly GithubApiFactory _githubApiFactory;
+    private readonly BbsApiFactory _bbsApiFactory;
+    private readonly EnvironmentVariableProvider _environmentVariableProvider;
+    private const int CHECK_STATUS_DELAY_IN_MILLISECONDS = 10000;
+
+    public MigrateRepoCommand(
+        OctoLogger log,
+        GithubApiFactory githubApiFactory,
+        BbsApiFactory bbsApiFactory,
+        EnvironmentVariableProvider environmentVariableProvider
+    ) : base("migrate-repo")
     {
-        private readonly OctoLogger _log;
-        private readonly GithubApiFactory _githubApiFactory;
-        private readonly EnvironmentVariableProvider _environmentVariableProvider;
+        _log = log;
+        _githubApiFactory = githubApiFactory;
+        _bbsApiFactory = bbsApiFactory;
+        _environmentVariableProvider = environmentVariableProvider;
+
+        Description = "Import a Bitbucket Server archive to GitHub.";
+        Description += Environment.NewLine;
+        Description += "Note: Expects GH_PAT env variable or --github-pat option to be set.";
 
-        public MigrateRepoCommand(
-            OctoLogger log,
-            GithubApiFactory githubApiFactory,
-            EnvironmentVariableProvider environmentVariableProvider
-        ) : base("migrate-repo")
+        // Arguments to generate a new archive
+        var bbsServerUrl = new Option<string>("--bbs-server-url")
         {
-            _log = log;
-            _githubApiFactory = githubApiFactory;
-            _environmentVariableProvider = environmentVariableProvider;
+            IsRequired = false,
+            Description = "The full URL of the Bitbucket Server/Data Center to migrate from."
+        };
 
-            Description = "Import a Bitbucket Server archive to GitHub.";
-            Description += Environment.NewLine;
-            Description += "Note: Expects GH_PAT env variable or --github-pat option to be set.";
+        var bbsProject = new Option<string>("--bbs-project")
+        {
+            IsRequired = false,
+            Description = "The Bitbucket project to import; defaults to '*' (all projects)."
```

Comment:
```suggestion
            Description = "The Bitbucket project to import."
```

I wouldn't mention the * syntax. In fact I would add validation code to ensure users DON'T pass * because that will break the octoshift backend ingestion I believe.